src/Adapter/Order/Refund/OrderSlipCreator.php

```diff
@@ -114,39 +114,41 @@ public function create(
 
             $customer = new Customer((int) $order->id_customer);
 
-            // @todo: use private method to send mail
-            $params = [
-                '{lastname}' => $customer->lastname,
-                '{firstname}' => $customer->firstname,
-                '{id_order}' => $order->id,
-                '{order_name}' => $order->getUniqReference(),
-            ];
-
-            $orderLanguage = $order->getAssociatedLanguage();
-
-            // @todo: use a dedicated Mail class (see #13945)
-            // @todo: remove this @and have a proper error handling
-            @Mail::Send(
-                (int) $orderLanguage->getId(),
-                'credit_slip',
-                $this->translator->trans(
-                    'New credit slip regarding your order',
-                    [],
-                    'Emails.Subject',
-                    $orderLanguage->locale
-                ),
-                $params,
-                $customer->email,
-                $customer->firstname . ' ' . $customer->lastname,
-                null,
-                null,
-                null,
-                null,
-                _PS_MAIL_DIR_,
-                true,
-                (int) $order->id_shop
-            );
-
+            if (!empty($customer->email)) {
```

Comment:
Hi @jf-viguier 
Could you please use `Validator::isLoadedObject` instead? I believe it's more appropriate here, because if the customer exists but the email is invalid, we should throw an exception. We need to clearly differentiate between the two cases: an invalid email and a deleted customer.

Reply:
Hi @PululuK I disagree, if the email in invalid, this has not to block refund action.
Here, we just send and email, it's a secondary action.